tests/std/tests/P2291R3_constexpr_integral_charconv/test.cpp

```diff
@@ -0,0 +1,428 @@
+        test_from_chars<T>("0000000000"sv, base, 10, errc{}, static_cast<T>(0));
+        test_from_chars<T>("000000000011"sv, base, 12, errc{}, static_cast<T>(base + 1));
+
+        // Test negative zero and negative leading zeroes.
+        if constexpr (is_signed_v<T>) {
+            test_from_chars<T>("-0"sv, base, 2, errc{}, static_cast<T>(0));
+            test_from_chars<T>("-0000000000"sv, base, 11, errc{}, static_cast<T>(0));
+            test_from_chars<T>("-000000000011"sv, base, 13, errc{}, static_cast<T>(-base - 1));
+        }
+
+        // N4713 23.20.3 [charconv.from.chars]/1 "The member ptr of the return value points to the
+        // first character not matching the pattern, or has the value last if all characters match."
+        test_from_chars<T>("11"sv, base, 2, errc{}, static_cast<T>(base + 1));
+        test_from_chars<T>("11@@@"sv, base, 2, errc{}, static_cast<T>(base + 1));
+
+        // When overflowing, we need to keep consuming valid digits, in order to return ptr correctly.
+        // This is really expensive so only test a few bases
+        if (base == 2 || base == 10 || base == 16 || base == 27 || base == 36) {
+            test_from_chars<T>(array_to_sv(hundred_ones), base, 100, out_ran);
+            test_from_chars<T>(array_to_sv(hundred_ones_and_atatat), base, 100, out_ran);
+
+            if constexpr (is_signed_v<T>) {
+                test_from_chars<T>(array_to_sv(minus_hundred_ones), base, 101, out_ran);
+                test_from_chars<T>(array_to_sv(minus_hundred_ones_and_atatat), base, 101, out_ran);
+            }
+        }
+    }
+
+    // N4713 23.20.3 [charconv.from.chars]/3 "The pattern is the expected form of the subject sequence
+    // in the "C" locale for the given nonzero base, as described for strtol"
+    // C11 7.22.1.4/3 "The letters from a (or A) through z (or Z) are ascribed the values 10 through 35"
+    for (int i = 0; i < 26; ++i) {
+        const char str1[] = {static_cast<char>('A' + i)};
+        const char str2[] = {static_cast<char>('a' + i)};
+        test_from_chars<T>(string_view(str1, 1), 36, 1, errc{}, static_cast<T>(10 + i));
+        test_from_chars<T>(string_view(str2, 1), 36, 1, errc{}, static_cast<T>(10 + i));
+    }
+
+    // N4713 23.20.3 [charconv.from.chars]/3 "no "0x" or "0X" prefix shall appear if the value of base is 16"
+    test_from_chars<T>("0x1729"sv, 16, 1, errc{}, static_cast<T>(0)); // reads '0', stops at 'x'
+    test_from_chars<T>("0X1729"sv, 16, 1, errc{}, static_cast<T>(0)); // reads '0', stops at 'X'
+
+    if constexpr (is_signed_v<T>) {
+        test_from_chars<T>("-0x1729"sv, 16, 2, errc{}, static_cast<T>(0)); // reads "-0", stops at 'x'
+        test_from_chars<T>("-0X1729"sv, 16, 2, errc{}, static_cast<T>(0)); // reads "-0", stops at 'X'
+    }
+}
+
+template <typename T>
+constexpr void test_integer() {
+    test_integer_to_chars<T>();
+    test_integer_from_chars<T>();
+}
+
+constexpr bool all_integer_tests() {

```

Comment:
We could probably get rid of the external `TEST_PART` machinery by splitting this function into `integer_tests_part0` and `integer_tests_part1`. After doing so, would it be possible to merge this back into the original `<charconv>` test? I hate all this duplication.